src/components/UserMarker.js

```diff
@@ -10,14 +10,21 @@ const UserMarker = () => {
 
   const map = useMap();
 
+  let positionSet = false;
```

Comment:
Using let like this is usually prone to bugs.
if the component rerenders it will lose its state. So usually use useState instead.